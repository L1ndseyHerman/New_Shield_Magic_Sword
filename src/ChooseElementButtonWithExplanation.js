import styles from './ChooseElementButtonWithExplanation.module.css';

function ChooseElementButtonWithExplanation(props)
{
    const buttonText = props.buttonText; 

    const chooseElementButtonPressed = () => {
       props.callback({pressedNextScreenButton: true, playerOneElement: buttonText}); 
    }

    return(
        <div className={styles.elementButtonDiv}>
            <div class="centeredDivPerfectButtonSize">
                <button style={{background: props.buttonColor}} 
                    onClick={() => {chooseElementButtonPressed()}}>{props.buttonText}</button>
            </div>
            <p>
                +1 damage to <span style={{color: props.explanationColor, 
                backgroundColor: "black"}}>{props.explanation}</span>.
            </p>
        </div>
    );
}
    
export default ChooseElementButtonWithExplanation;