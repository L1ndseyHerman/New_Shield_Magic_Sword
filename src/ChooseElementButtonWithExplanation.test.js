import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";

import renderer from 'react-test-renderer'

import ChooseElementButtonWithExplanation from "./ChooseElementButtonWithExplanation";

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it("renders explanation text", () => {

  //  For all of these tests, just testing the first (left-most) component, since the 
  //  other five should do sim things.
  act(() => {
    render(<ChooseElementButtonWithExplanation buttonText="Fire" explanation="Earth" />, container);
  });
  
  expect(container.querySelector("button").textContent).toBe("Fire");
  expect(container.querySelector("p").textContent).toBe("+1 damage to Earth.");

});

//  This isn't working, didn't notice when I changed the button color from "wheat" 
//  to "firebrick" :(
test('Snapshot of above test AND CSS', () => {
  const tree = renderer.create(<ChooseElementButtonWithExplanation />).toJSON()
  expect(tree).toMatchSnapshot();
})

/*test('it works', () => {
  const tree = renderer.create(<ChooseElementButtonWithExplanation />).toJSON()
  expect(tree).toHaveStyleRule('color', 'red')
  expect(tree).toHaveStyleRule('color', 'green', {
    media: '(max-width: 640px)',
    modifier: ':hover',
  })
})*/

