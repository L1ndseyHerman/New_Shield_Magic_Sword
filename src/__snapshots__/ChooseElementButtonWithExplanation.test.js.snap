// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of above test AND CSS 1`] = `
<div
  className="elementButtonDiv"
>
  <div
    class="centeredDivPerfectButtonSize"
  >
    <button
      onClick={[Function]}
      style={
        Object {
          "background": undefined,
        }
      }
    />
  </div>
  <p>
    +1 damage to 
    <span
      style={
        Object {
          "backgroundColor": "black",
          "color": undefined,
        }
      }
    />
    .
  </p>
</div>
`;
